documentation/nat-service/nat-service.md

```diff
@@ -0,0 +1,206 @@
+# NAT scenario
+
+## Overview
+
+NAT scenario decribes 2 NAT use cases:
+
+* VMs in VNET access Internet, aka SNAT.
+* Users from Internet access servcies hosted on the cloud VMs in VNET, aka DNAT.
+
+We use network appliance in this HLD, acting the role of NAT gateway as well as underlay router.
+This scenario provides the following capabilites:
+
+* SNAT service for TCP, UDP and ICMP packets.
+* DNAT service for TCP, UDP and ICMP packets.
+* Routing service for underlay forwarding.
+* Billing service(by traffic volumn and by commited traffic rate).
+
+The goal is to test the following properties:
+
+* Connection set up rate in Connections per Second (CPS)
+* Data traffic bandwidth in Bits per Second (BPS)
+* Data traffic performance in Packet per Second (PPS)
+* Connection Flow Scale
+
+## NAT scenario architecture
+
+![nat-architecture](images/Figure1-dash-nat-topo.png)
+*Figure 1 - NAT scenario architecture*
+
+Tenant1 access Internet service, with VM1 in VNET1 through NAT gateway running on a smart switch. Tenant2 provide service to the public, hosted on VM2 in VNET2, Internet user access this service through NAT gateway running on a smart switch.
+
+For both case, VMs in different VNET connects to Internet through a NAT gateway.
+
+## Packet proccessing pipeline in NAT scenario
+
+### 1. SNAT scenario
+
+#### 1.1 Outbound packet processing pipeline
+
+![snat-outbound-pipeline](images/Figure2-snat-outbound-pipeline.png)
+*Figure 2 - SNAT outbound packet processing pipeline*
+
+* The pipeline determines the **outbound direction** based on the presence of a VXLAN header in the incoming packet.  The tenant's VNI is a part of the key in flow table entries.
```

Comment:
The presence or absence of the vxlan header can be used to distinguish between SNAT vs DNAT can work as long as those are the only two services running on this box. It could become an issue in future if other services need to be offloaded on the same box.